### PR TITLE
[SMD-324] Front end pre-validation summary changes

### PR DESCRIPTION
**Pull request by Connor-OS**
Changed the presentation of pre-validation messages to use error summary component and bulleted lists of error messages.

### Change description
_A brief description of the pull request_

- [ ] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [ ] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
_If manual testing is needed, give suggested testing steps_

### Screenshots of UI changes (if applicable)
image

### CODE REVIEW

File: app/templates/main/upload.html [hunk 1]
```diff
@@ -94,20 +100,26 @@ <h3 class="govuk-heading-s">Re-upload your data return</h3>
...
-                        {% if not pre_error %}
-                            <h2 class="govuk-heading-m">Upload your data return</h2>
+                        <h2 class="govuk-heading-m">Upload your data return</h2>
+                        {% if pre_error %}
+                            {# construct html to display errors in bullet point list #}
+                            {% set ns = namespace(pre_error_html = "<div class=\"govuk-error-message\">Some of your data is not what we expect, so we could not run the full checks.
+                            Fix these problems and re-upload:") %}
+                            {% set ns.pre_error_html = ns.pre_error_html ~ "<ul>" %}
+                            {% for error in pre_error %}
+                                {% set ns.pre_error_html = ns.pre_error_html ~ "<li>" ~ error ~ "</li>" %}
```
[the remark this answers is not in the document]
Comment (Connor-OS) [reply app/templates/main/upload.html:110]: ~ is a jinja specific operator for string concatination. I'll remove it becasue it'ssuper confusing and + works just as well

[the remark this answers is not in the document]
Comment (Connor-OS) [reply app/templates/main/upload.html:110]: Error summory does contain an anchor link in the errorSummary macro above

**Comment by cyrusdobbs** [comment]
LGTM - only thing is should the messages at the top have bullet points?

**Review by cyrusdobbs — APPROVED** [review]
LGTM

**Pull request merged**